### PR TITLE
Feature/14512 user service update

### PR DESCRIPTION
**Pull request by GuyHarwood**
Better structure to promises and tests

### CODE REVIEW

File: pupil-spa/src/app/user.service.ts [hunk 1]
```diff
@@ -2,48 +2,47 @@ import { Injectable } from '@angular/core';
...
-      if (err) {
-        return reject(err);
-      }
-      if (data) {
-        return resolve(data);
-      }
+          const data = response.json();
+          this.loggedIn = true;
+          this.storageService.setItem(sessionDataKey, data['session']);
+          this.storageService.setItem(questionsDataKey, data['questions']);
```
Comment (jon-shipley) [inline pupil-spa/src/app/user.service.ts]: I think we also need to set the config section too?

File: pupil-spa/src/app/user.service.ts [hunk 2]
```diff
@@ -2,48 +2,47 @@ import { Injectable } from '@angular/core';
...
+        (err) => {
+          reject(err);
+        }).catch(error => reject(error));
     });
   }
 
   logout() {
-    this.storageService.removeItem(auth_token);
-    this.storageService.removeItem('data');
+    /*   this.storageService.removeItem(auth_token);
```
Comment (jon-shipley) [inline pupil-spa/src/app/user.service.ts]: Why are these now commented out?  Aren't they needed?

**Comment by jon-shipley** [comment]
Please merge the latest master in and resolve the conflicts.
.

File: pupil-spa/src/app/user.service.ts (hunk 1 above)
Comment (GuyHarwood) [inline pupil-spa/src/app/user.service.ts]: yes, its not in the checkSessionSchema.json, so will add that and the assertion

File: pupil-spa/src/app/user.service.ts (hunk 2 above)
Comment (GuyHarwood) [inline pupil-spa/src/app/user.service.ts]: good catch.  I hadn't saved the file!

**Pull request merged**